runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkJobInvocation.java

```diff
@@ -98,41 +99,71 @@ private FlinkJobInvocation(
   private PipelineResult runPipeline() throws Exception {
     MetricsEnvironment.setMetricsSupported(false);
 
+    FlinkPortablePipelineTranslator<?> translator;
+    if (!pipelineOptions.isStreaming() && !hasUnboundedPCollections(pipeline)) {
+      // TODO: Do we need to inspect for unbounded sources before fusing?
```

Comment:
Do we even have a way to inspect this for portable pipelines? We don't really know whether a transform is bounded or not.

Reply:
Ah, I saw you copied this over.